syntaxes/pact.tmLanguage.json

```diff
@@ -1,347 +1,350 @@
+				"beginCaptures": {
+				  "1": {
+					"name": "variable.let.pact"
+				  }
+				},
+				"end": "\\)",
+				"name": "meta.let.binding.pact",
+				"patterns": [
+				  {
+					"include": "$self"
+				  }
+				]
+			  }
+			]
+		  }
+		]
+	  },
+	  "sexp": {
+		"begin": "(\\()",
+		"beginCaptures": {
+		  "1": {
+			"name": "punctuation.section.expression.begin.pact"
+		  }
+		},
+		"end": "(\\))(\\n)|(\\)(?=[\\}\\]\\)\\s]*(?:;|$)))|(\\))",
+		"endCaptures": {
+		  "1": {
+			"name": "punctuation.section.expression.end.trailing.pact"
+		  },
+		  "2": {
+			"name": "meta.after-expression.pact"
+		  },
+		  "3": {
+			"name": "punctuation.section.expression.end.trailing.pact"
+		  },
+		  "4": {
+			"name": "punctuation.section.expression.end.pact"
+		  }
+		},
+		"name": "meta.expression.pact",
+		"patterns": [
+		  {
+			"include": "#deflam"
+		  },
+		  {
+			"include": "#defsimple"
+		  },
+		  {
+			"include": "#string"
+		  },
+		  {
+			"include": "#reserved"
+		  },
+		  {
+			"include": "#literal"
+		  },
+		  {
+			"include": "#list"
+		  },
+		  {
+			"include": "#object"
+		  },
+		  {
+			"include": "#let"
+		  },
+		  {
+			"include": "#sexp"
+		  },
+		  {
+			"match": "(?<=\\()(.+?)(?=\\s|\\))",
+			"captures": {
+			  "1": {
+				"name": "entity.name.function.pact"
+			  }
+			},
+			"patterns": [
+			  {
+				"include": "$self"
+			  }
+			]
+		  },
+		  {
+			"include": "$self"
+		  }
+		]
+	  },
+	  "deflam": {
+		"begin": "(?<=\\()(defun|defpact|defcap)\\s+([\\w%#+\\-_&\\$@<>=\\?\\*!\\|/]+)",
+		"beginCaptures": {
+		  "1": {
+			"name": "keyword.reserved.pact"
+		  },
+		  "2": {
+			"name": "entity.function.name.pact"
+		  }
+		},
+		"name": "meta.definition.global.pact",
+		"end": "\\)",
+		"patterns": [
+		  {
+			"include": "#arglist"
+		  },
+		  {
+			"include": "#type"
+		  }
+		]
+	  },
+	  "arglist": {
+		"begin": "(?<=\\()",
+		"end": "(?=\\))",
+		"patterns": [
+		  {
+			"match": "([\\w%#+\\-_&\\$@<>=\\?\\*!\\|/]+)",
+			"name": "variable.name.arg.pact"
+		  },
+		  {
+			"include": "#type"
+		  }
+		]
+	  },
+	  "defsimple": {
+		"begin": "(?<=\\()(defconst|defschema|deftable|module|interface)\\s+([\\w%#+-_&\\$@<>=\\?\\*!\\|/]+)\\b",
+		"beginCaptures": {
+		  "1": {
+			"name": "keyword.reserved.pact"
+		  },
+		  "2": {
+			"name": "entity.function.name.pact"
+		  }
+		},
+		"end": "(?=\\))",
+		"name": "meta.definition.global.pact",
+		"patterns": [
+		  {
+			"include": "$self"
+		  }
+		]
+	  },
+	  "string": {
+		"begin": "(?<!\\\\)(\")",
+		"beginCaptures": {
+		  "1": {
+			"name": "punctuation.definition.string.begin.pact"
+		  }
+		},
+		"end": "(?<!\\\\)(\")",
+		"endCaptures": {
+		  "1": {
+			"name": "punctuation.definition.string.end.pact"
+		  }
+		},
+		"name": "string.quoted.double.pact"
+	  },
+	  "atom": {
+		"patterns": [
+		  {
+			"match": "([\\w%#+-_&\\$@<>=\\?\\*!\\|/]+)"
+		  }
+		]
+	  },
+	  "symbol": {
+		"patterns": [
+		  {
+			"begin": "'",
+			"end": "(?=\\s|[],:\"'.)])",
```

Comment:
This regex doesn't seem correct to me, it has a closing parenthesis to many. Did you intend the closing parenthesis to be escaped?

Reply:
@EnoF it is correct.
"You can include an unescaped closing bracket by placing it right after the opening bracket, or right after the negating caret. []x] matches a closing bracket or an x." source: https://www.regular-expressions.info/charclass.html

Reply:
The Pact extension for Atom had initially the same issue, let me refer this earlier exchange:
https://github.com/kadena-io/pact-atom/pull/9#discussion_r735581387

Reply:
The ultimate review of the actual functionality is of course building the extension at the specific point commit using VSCE command and install it locally in VS Code.